tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_remainder.h

```diff
@@ -0,0 +1,68 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+#include "noc_nonblocking_api.h"
+#include "ckernel_sfpu_converter.h"
+#include "ckernel_sfpu_recip.h"
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+template <bool APPROXIMATION_MODE, int ITERATIONS = 8>
+inline void calculate_remainder(const uint value, const uint recip) {
+
+    // SFPU microcode
+    Converter c_value;
+    c_value.u = value;
+    vFloat s = c_value.f;
+    vFloat value_tmp = s;
+    s = sfpi::abs(s);
+
+    c_value.u = recip;
+    vFloat recip_val = c_value.f;
+    recip_val = sfpi::abs(recip_val);
+
+    #pragma GCC unroll 0
+    for (int d = 0; d < ITERATIONS; d++) {
+        vFloat val = dst_reg[0];
+        vFloat v = sfpi::abs(val);
+
+        vFloat quotient = v*recip_val;
+
+        vInt tmp = float_to_int16(quotient);
```

Comment:
Can you leave a `//TODO` note here that once float to int32 cast is available, this implementation should switch to using that?

Reply:
Sure Reem will add TODO here